### PR TITLE
add 'current_droplet' to app's relationships

### PR DESCRIPTION
**Pull request by gmllt**
The only way to retrieve the current droplet of an application via the API is to go through the `/v3/apps/:app_guid/droplets/current` or `/v3/apps/:app_guid/relationships/current_droplet` endpoints.

In a context where we want to retrieve all the applications (more than 3000 in my case) and their current droplet, it is therefore necessary to make one request per application (more than 3000 requests).

This PR aims to make the current_droplet relationship visible on all endpoints presenting the app object : 
- `/v3/apps`
- `/v3/apps/:app_guid`
- `/v3/stacks/:stack_id/apps`
- ...

See issue #3860 

* [x] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/CONTRIBUTING.md)

* [x] I have viewed, signed, and submitted the Contributor License Agreement

* [x] I have made this pull request to the `main` branch

* [x] I have run all the unit tests using `bundle exec rake`

* [x] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/spec/README.md#cf-acceptance-tests-cats)

### CODE REVIEW

File: docs/v3/source/includes/resources/apps/_object.md.erb [hunk 1]
```diff
@@ -16,6 +16,7 @@ Name | Type | Description
 **state** | _string_ | Current desired state of the app; valid values are `STOPPED` or `STARTED`
 **lifecycle** | [_lifecycle object_](#the-lifecycle-object) | Provides the default lifecycle object for the application. This lifecycle will be used when staging and running the application. The staging lifecycle can be overridden on [builds](#builds)
 **relationships.space** | [_to-one relationship_](#to-one-relationships) | The space the app is contained in
+**relationships.droplet** | [_to-one relationship_](#to-one-relationships) | The current droplet use by the application
```
[the remark this answers is not in the document]
Comment (gmllt) [reply docs/v3/source/includes/resources/apps/_object.md.erb]: thanks, i missed it

File: spec/unit/presenters/v3/app_presenter_spec.rb [hunk 2]
```diff
@@ -52,6 +52,7 @@ module VCAP::CloudController::Presenters::V3
         expect(result[:lifecycle][:data][:stack]).to eq('the-happiest-stack')
         expect(result[:lifecycle][:data][:buildpacks]).to eq(['git://***:***@github.com/repo', 'limabean'])
         expect(result[:relationships][:space][:data][:guid]).to eq(app.space.guid)
+        expect(result[:relationships][:current_droplet][:data][:guid]).to be_nil or eq(app.droplet.guid)
```
[the remark this answers is not in the document]
Comment (gmllt) [reply spec/unit/presenters/v3/app_presenter_spec.rb]: thanks for the advice, it should be ok now

File: spec/request/apps_spec.rb [hunk 3]
```diff
@@ -2283,6 +2325,11 @@
                 'data' => {
                   'guid' => space.guid
                 }
+              },
+              'current_droplet' => {
+                'data' => {
+                  'guid' => app_model.droplet.guid
```
[the remark this answers is not in the document]
Comment (gmllt) [reply spec/request/apps_spec.rb]: thanks, i've switched to droplet.guid

File: spec/request/apps_spec.rb [hunk 4]
```diff
@@ -2617,6 +2664,11 @@
               'data' => {
                 'guid' => space.guid
               }
+            },
+            'current_droplet' => {
+              'data' => {
+                'guid' => droplet_guid_regex
```
[the remark this answers is not in the document]
Comment (gmllt) [reply spec/request/apps_spec.rb]: switched to use droplet.guid

File: spec/request/apps_spec.rb [hunk 5]
```diff
@@ -2243,6 +2284,7 @@
         desired_state: 'STOPPED'
       )
     end
+    let(:droplet_guid_regex) { /^[a-f0-9]{8}(-[a-f0-9]{4}){4}[a-f0-9]{8}$/ }
```
[the remark this answers is not in the document]
Comment (gmllt) [reply spec/request/apps_spec.rb]: removed as it's not used anymore

File: spec/request/apps_spec.rb [hunk 6]
```diff
@@ -1284,6 +1308,7 @@
       app_model.lifecycle_data.buildpacks = [buildpack.name]
       app_model.lifecycle_data.stack = stack.name
       app_model.lifecycle_data.save
+      app_model.droplet_guid = 'a-droplet-guid'
```
[the remark this answers is not in the document]
Comment (gmllt) [reply spec/request/apps_spec.rb]: removed

File: spec/request/apps_spec.rb [hunk 7]
```diff
@@ -11,6 +11,7 @@
   let(:stack) { VCAP::CloudController::Stack.make }
   let(:user_email) { Sham.email }
   let(:user_name) { 'some-username' }
+  let(:droplet_guid_regex) { /^[a-f0-9]{8}(-[a-f0-9]{4}){4}[a-f0-9]{8}$/ }
```
[the remark this answers is not in the document]
Comment (gmllt) [reply spec/request/apps_spec.rb]: thanks, i've removed it since i use droplet.guid instead

**Review by philippthun — APPROVED** [review]

**Pull request merged**